torch/__init__.py

```diff
@@ -45,26 +45,32 @@
     py_dll_path = os.path.join(sys.exec_prefix, 'Library', 'bin')
     th_dll_path = os.path.join(os.path.dirname(__file__), 'lib')
 
-    if not os.path.exists(os.path.join(th_dll_path, 'nvToolsExt64_1.dll')) and \
-            not os.path.exists(os.path.join(py_dll_path, 'nvToolsExt64_1.dll')):
+    if sys.exec_prefix != sys.base_exec_prefix:
+        base_py_dll_path = os.path.join(sys.base_exec_prefix, 'Library', 'bin')
+    else:
+        base_py_dll_path = ''
+
+    dll_paths = list(filter(os.path.exists, [th_dll_path, py_dll_path, base_py_dll_path]))
+
+    if all(map(lambda p: not os.path.exists(os.path.join(p, 'nvToolsExt64_1.dll')), dll_paths)):
         nvtoolsext_dll_path = os.path.join(
             os.getenv('NVTOOLSEXT_PATH', 'C:\\Program Files\\NVIDIA Corporation\\NvToolsExt'), 'bin', 'x64')
     else:
         nvtoolsext_dll_path = ''
 
     from .version import cuda as cuda_version
     import glob
-    if cuda_version and len(glob.glob(os.path.join(th_dll_path, 'cudart64*.dll'))) == 0 and \
-            len(glob.glob(os.path.join(py_dll_path, 'cudart64*.dll'))) == 0:
+    if cuda_version and all(map(lambda p: len(glob.glob(os.path.join(p, 'cudart64*.dll'))) == 0, dll_paths)):
```

Comment:
Nit (if you want to make this statement even more compact, you can change `len(foo) == 0` to `not foo` (empty lists are false in Python, aren't they?)
```suggestion
    if cuda_version and all([len(glob.glob(os.path.join(p, 'cudart64*.dll'))) == 0 for p in dll_paths]):
```